lib/ipam/ipam.go

```diff
@@ -239,8 +252,15 @@ func (c ipamClient) determinePools(requestedPoolNets []net.IPNet, version int) (
 }
 
 func (c ipamClient) autoAssign(ctx context.Context, num int, handleID *string, attrs map[string]string, requestedPools []net.IPNet, version int, host string, maxNumBlocks int) ([]net.IP, error) {
+	// Retrieve node for given hostname to use for ip pool node selection
+	node, err := c.client.Get(ctx, model.ResourceKey{Kind: v3.KindNode, Name: host}, "")
+	if err != nil {
+		log.WithError(err).WithField("host", host).Warn("failed to get node for host")
```

Comment:
Should upgrade this log message to be error level

Reply:
Also, the IPAM code calls this "host" but everywhere else we refer to this as "node" (it's an historical thing that the IPAM code still calls this host).

So, let's just use the terminology "node" instead of "host" in all of the logging to be consistent.